### PR TITLE
XCTest Link Errors and Unresolved struct access control issues 

### PR DESCRIPTION
**Pull request by alldritt**
I ran into problems linking a test app where XCTest was unresolved.  I also was not able to access the properties of the Unresolved struct as described in the README file.  Hopefully these changes are aligned with what you intended.

### CODE REVIEW

File: Sources/MathParser/MathParser.swift [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // Copyright © 2021 Brad Howes. All rights reserved.
 
 import Parsing
-import XCTest
+import Foundation
+import XCTestDynamicOverlay
```
Comment (bradhowes) [inline Sources/MathParser/MathParser.swift]: I don't think this is necessary after swapping XCTest for Foundation.

**Review by bradhowes — CHANGES_REQUESTED** [review]
Thanks so much for the PR!
Just one minor tweak and should be good to go.

File: Sources/MathParser/MathParser.swift (hunk 1 above)
Comment (alldritt) [reply Sources/MathParser/MathParser.swift]: Without Foundation many standard library symbols are undefined: sin, cos, etc.

Comment (alldritt) [reply Sources/MathParser/MathParser.swift]: If I restore XCTest in place of Foundation then XCTest becomes an undefined symbol.

Perhaps I'm missing something in the test application I'm using.  You can see my work here:

https://github.com/alldritt/MathParserTest.git

Comment (bradhowes) [reply Sources/MathParser/MathParser.swift]: I meant you are correct to do what you did but there is no reason to add the overlay import.

Comment (bradhowes) [reply Sources/MathParser/MathParser.swift]: I check out your project and everything will work just fine all of your changes, but you can remove the "import XCTestDynamicOverlay" -- it is not necessary. I can remove it after or just update the PR with a new commit.

Comment (bradhowes) [reply Sources/MathParser/MathParser.swift]: BTW, nice to meet you! You've helped me out a ton with your debugger.

Capto_Capture 2023-04-19_08-55-23_PM

**Review by bradhowes — APPROVED** [review]
I went ahead and removed the needless import. Thanks again!

**Pull request merged**